openid-zkp-auth/circuits/jwt_proof_ua.circom

```diff
@@ -0,0 +1,170 @@
+pragma circom 2.1.3;
+
+include "helpers/sha256.circom";
+include "helpers/misc.circom";
+include "helpers/strings.circom";
+include "helpers/zkhasher.circom";
+
+/**
+JWT Proof: User-agnostic (UA) circuit
+
+    Construction params:
+    - maxContentLength:         Maximum length of the JWT + SHA2 padding in bytes
+    - maxSubLength:             Maximum length of the subject_id (in ascii)
+
+    Private Inputs:
+    - content[inCount]:         Segments of X as inWidth bit chunks where X is JWT header + JWT payload + SHA-2 padding + zeroes
+    - sub_length_ascii:         Length of the subject_id in ASCII, e.g., for ',"sub":12345,' it is 13
+    - subject_id[maxSubLength]: The subject (user) ID for the first sub_length_ascii characters and 0s for the rest
+    - subject_pin:              User's PIN to keep the subject_id private
+    - mask[inCount]:            A binary mask over X, i.e., mask[i] = 0 or 1
+    - randomness:               A 248-bit random number to keep the sensitive parts of JWT hidden
```

Comment:
based on the code below, `nonce = H(pk | last epoch | randomness)`

if so,
1. please add this to the comment
2. why 248 bit? 128 is enough for security here
3. `nonce` is expected to be revealed to the verifier? I assume that not, since we depend on its entropy to hide sensitive fields as described here. However, below `nonce` is included in `all_inputs_hash` - how can the verifier check that?
I'd guess that `all_inputs_hash` should only include the eph_pk and max epoch, no?

Reply:
Fixed